python/cugraph/cugraph/tests/internals/test_replicate_edgelist_mg.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+
+import pytest
+
+import dask_cudf
+import numpy as np
+from cugraph.testing import UNDIRECTED_DATASETS, karate_disjoint
+
+from cugraph.structure.replicate_edgelist import replicate_edgelist
+from cudf.testing.testing import assert_frame_equal
+from pylibcugraph.testing.utils import gen_fixture_params_product
+
+
+# =============================================================================
+# Pytest Setup / Teardown - called for each test function
+# =============================================================================
+def setup_function():
+    gc.collect()
+
+
+edgeWeightCol = "weights"
+edgeIdCol = "edge_id"
+edgeTypeCol = "edge_type"
+srcCol = "src"
+dstCol = "dst"
+
+
+input_data = UNDIRECTED_DATASETS
+input_data.append(karate_disjoint)
```

Comment:
You're modifying the `UNDIRECTED_DATASETS` list for every test in the pytest session here.  I think you really want to do this?
```suggestion
input_data = UNDIRECTED_DATASETS + [karate_disjoint]
```